src/jollyjumper/rules.py

```diff
@@ -14,10 +24,158 @@ def get_sirrematic_enjambment(previous_token, next_token):
+        return [previous_token.pos_, next_token.pos_]
+    return None
+
+
+def get_sirrematic_with_verb_enjambment(previous_token, next_token):
+    """
+    Checks if sirrematic enjambment exists between two lines
+    :param previous_token: The word before a newline character
+    :param next_token: The word before a newline character
+    :return: Sirrematic type or None if not found
+    """
+    if previous_token.pos_ in ('AUX', 'VERB') and next_token.pos_ == 'VERB':
+        return [previous_token.pos_, next_token.pos_]
+    return None
+
+
+def get_sirrematic_relation_words_verbs_enjambment(previous_token, next_token):
+    """
+    Checks if sirrematic enjambment exists between two lines
+    :param previous_token: The word before a newline character
+    :param next_token: The word before a newline character
+    :return: Sirrematic type or None if not found
+    """
+    if previous_token.pos_ in ('AUX', 'VERB') and re.search('AdpType=Prep',
+                                                            next_token.tag_) \
+            and next_token.lower_ in ('de', 'del'):
+        return [previous_token.pos_, 'PREP']
+    return None
+
+
+def get_sirrematic_orational_enjambment(previous_token, next_token):
+    """
+    Checks if sirrematic enjambment exists between two lines
+    :param previous_token: The word before a newline character
+    :param next_token: The word before a newline character
+    :return: Sirrematic type or None if not found
+    """
+    if previous_token.pos_ in ('ADJ', 'NOUN', 'ADV') \
+            and re.search('NumType', next_token.tag_) \
+            and next_token.lower_ in (
+            'que', 'cuyo', 'cuya', 'cuyos', 'cuyas', 'donde'):
+        return [previous_token.pos_, next_token.pos_]
+    return None
+
+
+def get_sirrematic_prepositional_enjambment(previous_token, next_token):
+    """
+    Checks if sirrematic enjambment exists between two lines
+    :param previous_token: The word before a newline character
+    :param next_token: The word before a newline character
+    :return: Sirrematic type or None if not found
+    """
+    if next_token.n_rights > 0:
+        if previous_token.pos_ in ('ADJ', 'ADV', 'NOUN') and re.search(
+                'AdpType=Prep', next_token.tag_) \
+                and next_token.nbor().is_ancestor(previous_token):
+            return [previous_token.pos_, 'PREP']
+    return None
+
+
+def get_sirrematic_prepositional_without_de_enjambment(previous_token,
+                                                       next_token):
+    """
+    Checks if sirrematic enjambment exists between two lines
+    :param previous_token: The word before a newline character
+    :param next_token: The word before a newline character
+    :return: Sirrematic type or None if not found
+    """
+    if next_token.n_rights > 0:
+        if previous_token.pos_ in ('ADJ', 'NOUN') \
+                and re.search('AdpType=Prep', next_token.tag_) \
```

Comment:
Maybe better between parenthesis. Plus, the first two `if`s could be merged into one.